integration-tests/tests/utils/aws/s3GetAvailableTestDate.ts

```diff
@@ -0,0 +1,50 @@
+import { ListObjectsCommand } from '@aws-sdk/client-s3'
+import { getEnv } from '../helpers'
+import { s3Client } from './s3Client'
+
+export const getAvailableTestDate = async () => {
+  return findAvailableS3Locations(generateRandomDateAndPrefix())
+}
+
+const findAvailableS3Locations = async (checkDate: MatchedDateAndPrefix) => {
+  const buckets = [getEnv('ANALYSIS_BUCKET_NAME'), getEnv('AUDIT_BUCKET_NAME')]
+
+  const locationAvailability = await Promise.all(
+    buckets.map(async (bucket) => {
+      const input = {
+        Bucket: bucket,
+        Prefix: checkDate.prefix
+      }
+      const command = new ListObjectsCommand(input)
+      const response = await s3Client.send(command)
+
+      return !response.Contents || response.Contents.length === 0 ? true : false
+    })
+  )
+
+  if (locationAvailability.includes(false)) {
+    await findAvailableS3Locations(generateRandomDateAndPrefix())
```

Comment:
I would feel less queasy about this if it had some kind of maximum number of retries.